.github/workflows/ci-uxd.yml

```diff
@@ -0,0 +1,143 @@
+          path: ./uxd-program
+
+      - uses: actions/setup-node@v2.5.1
+        name: Node 14 setup
+        with:
+          node-version: "14"
+          cache: "yarn"
+          cache-dependency-path: ./uxd-program/yarn.lock
+          registry-url: 'https://npm.pkg.github.com'
+          scope: "@UXDProtocol"
+          always-auth: true
+
+      - uses: actions/cache@v2
+        name: Cache Cargo registry + index.
+        id: cache-anchor
+        with:
+          path: |
+            ~/.cargo/bin/
+            ~/.cargo/registry/index/
+            ~/.cargo/registry/cache/
+            ~/.cargo/git/db/
+          key: cargo-${{ runner.os }}-v0000-${{ hashFiles('**/Cargo.lock') }}
+                
+      - name: Add Solana to PATH
+        run: echo "/home/runner/.local/share/solana/install/active_release/bin:$PATH" >> $GITHUB_PATH
+        
+      - uses: actions/cache@v2
+        name: Cache Solana Tool Suite
+        id: cache-solana
+        with:
+          path: |
+            ~/.cache/solana/
+            ~/.local/share/solana/
+          key: solana-${{ runner.os }}-v0000-${{ env.solana_version }}
+
+      - uses: actions/cache@v2
+        name: Cache target folder
+        id: cache-programs
+        with:
+          path: |
+            ./target/debug
+            ./target/release
+            ./target/bpfel-unknown-unknown
+          key: solana-${{ runner.os }}-v0000-${{ hashFiles('**/programs/**/*.rs') }}-${{ hashFiles('**/programs/**/*.toml') }}
+
+      - uses: actions-rs/toolchain@v1
+        name: Rust toolchain
+        with:
+          toolchain: nightly
+          override: true
+
+      - name: Install solana
+        if: steps.cache-solana.outputs.cache-hit != 'true'
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y pkg-config build-essential libudev-dev
+          sh -c "$(curl -sSfL https://release.solana.com/v${{ env.solana_version }}/install)"
+
+      - name: Setup Solana
+        run: |
+          solana --version
+          solana config set -u d
+          solana config get
+          solana-keygen new -o ${{ env.runner_wallet_path }}
+          
+      - name: Checkout MangoClient-v3, install dependencies, and run MM bot devnet SOL perp + Keeper
+        uses: actions/checkout@v2.4.0
+        with:
+          repository: blockworks-foundation/mango-client-v3
+          path: ./mango-client-v3
+
+      - name: Install MangoClient-v3 dependencies then runs Mango Keeper and MM bot on devnet for SOL perp
+        run: |
+          cd ./mango-client-v3
+          yarn install
+          solana airdrop 1 -u d
+          solana airdrop 1 -u d
+          solana airdrop 1 -u d
+          solana airdrop 1 -u d
+          solana airdrop 1 -u d
+          solana airdrop 1 -u d
+          solana airdrop 1 -u d
+          solana airdrop 1 -u d
+          solana airdrop 1 -u d
+          solana airdrop 1 -u d
+          solana airdrop 1 -u d
+          solana airdrop 1 -u d
```

Comment:
nitpick: can one technically send the sol back to the faucet? - be a good citizen and all